### ISSUE TITLE
fix convoluted workflow for PRing changes that require new pip wheels

### ISSUE THREAD

**Issue by RussTedrake** [RussTedrake/manipulation#395]
- We currently have some CI (in the noncritical.yaml) which test the current repo against the deployed pip wheels).
- We also have a `check_deepnote_requirements` test in the repo that will fail (in the main ci) if the `version` in `pyproject.toml` doesn't match the pip wheels deployed on deepnote.

I should be able to fix this so that local changes can pass the tests, and that CI can warn if a change requires a new pip wheel to be deployed (both in PR and in nightlies).

The current recommended (but definitely suboptimal) workflow is:
1) PR changes _without_ changing the version in pyproject.toml and _without_ deploying the new pip wheel.  Accept that the noncritical CI may fail. (e.g. if the examples in master no longer work with the pip version of this repo).
2) Once it lands in master, then update the version, push the pip wheels (and deepnote references) and then push directly to master; make sure all tests pass. The post-merge CI will complain if you've gotten it wrong.

The obvious problem is that we could push something broken to master. But the current alternative -- having the pip wheels generated from an open, unmerged, PR seems worse to me.

**Comment by nepfaff** [comment]
I have been using the "current alternative" approach so far just before merging to avoid pushing a broken branch. I do agree that this isn't great but I'm not a fan of the first approach either.

An easy workaround for now would be to move the `check_deepnote_requirements` into a separate CI, preventing it from hiding other test failures. I can quickly do that if you agree.
I'm guessing that you mean something like this when saying "that CI can warn if a change requires a new pip wheel to be deployed"? From a quick look, there are no great "warning" type failures that have good visibility.

**Comment by RussTedrake** [comment]
> move the check_deepnote_requirements into a separate CI

Yes, I think this is a strict improvement. But the way that I would do this would be to disable that test via an exclusion (e.g. a bazel test tag) in the instances of CI that shouldn't have to check it.  I would still want my `bazel test //...` to check it locally.

**Comment by nepfaff** [comment]
The conditional execution solution is great! I didn't know this was an option
